src/videoCache.js

```diff
@@ -162,6 +170,36 @@ export function getCacheUrl(id) {
   return `${config.getConfig('cache.url')}?uuid=${id}`;
 }
 
+export const storeLocally = (bid) => {
+  const vastValue = getVastValue(bid);
+  const dataUri = 'data:text/xml;base64,' + btoa(vastValue);
+  bid.vastUrl = dataUri;
+  //@todo: think of wrapping it with [if (adServer)]
+  vastsLocalCache.set(getLocalCacheBidId(bid), dataUri);
+};
+
+export async function getLocalCachedBidWithGam(adTagUrl) {
+  const gamAdTagUrl = new URL(adTagUrl);
+  const custParams = new URLSearchParams(gamAdTagUrl.searchParams.get('cust_params'));
+  const hb_bidder = custParams.get('hb_bidder');
+  const hb_adid = custParams.get('hb_adid');
+  const response = await fetch(gamAdTagUrl);
+
+  if (!response.ok) {
+    logError('Unable to fetch valid response from Google Ad Manager');
+    return;
+  }
+
+  const gamVastWrapper = await response.text();
+  const bidVastDataUri = vastsLocalCache.get(`${hb_bidder}_${hb_adid}`);
+  const mockUrl = LOCAL_CACHE_MOCK_URL + hb_bidder;
```

Comment:
the "normal" cache generates a UUID which is then passed on to the creative and echoed back in the VAST.

I think the local cache could also generate a UUID, pass it along in the same way, and this could look for it in the response contents instead of imposing a different URL structure in the creative. It'd have the advantage of working with existing integrations, and being easier to switch into / out of.